test/test_char_array.cpp

```diff
@@ -14,44 +14,75 @@
 
 #include <gtest/gtest.h>
 
+#include "./allocator_testing_utils.h"
 #include "rcutils/allocator.h"
-
+#include "rcutils/error_handling.h"
 #include "rcutils/types/char_array.h"
 
-TEST(test_char_array, default_initialization) {
-  auto char_array = rcutils_get_zero_initialized_char_array();
+class ArrayCharTest : public ::testing::Test
+{
+protected:
+  void SetUp() override
+  {
+    // Reset rcutils error to prevent failure from a previous test case to contaminate
+    // another one.
+    rcutils_reset_error();
+
+    allocator = rcutils_get_default_allocator();
+    char_array = rcutils_get_zero_initialized_char_array();
+  }
+
+  rcutils_allocator_t allocator;
+  rcutils_char_array_t char_array;
+};
+
+static rcutils_ret_t example_logger(
+  rcutils_char_array_t * char_array,
+  const char * format, ...)
+{
+  rcutils_ret_t status;
+  va_list args;
+  va_start(args, format);
+  status = rcutils_char_array_vsprintf(char_array, format, args);
+  va_end(args);
+  return status;
+}
 
-  auto allocator = rcutils_get_default_allocator();
+TEST_F(ArrayCharTest, default_initialization) {
   EXPECT_EQ(RCUTILS_RET_OK, rcutils_char_array_init(&char_array, 0, &allocator));
   EXPECT_EQ(0lu, char_array.buffer_capacity);
   EXPECT_EQ(RCUTILS_RET_OK, rcutils_char_array_fini(&char_array));
   EXPECT_EQ(0lu, char_array.buffer_capacity);
   EXPECT_FALSE(char_array.buffer);
 }
 
-TEST(test_char_array, resize) {
-  auto char_array = rcutils_get_zero_initialized_char_array();
-  auto allocator = rcutils_get_default_allocator();
-  auto ret = rcutils_char_array_init(&char_array, 5, &allocator);
+TEST_F(ArrayCharTest, resize) {
+  rcutils_ret_t ret = rcutils_char_array_init(&char_array, 5, &allocator);
   ASSERT_EQ(RCUTILS_RET_OK, ret);
 
-  memcpy(char_array.buffer, "1234\0", 5);
+  memcpy(char_array.buffer, "1234", 5);
```

Comment:
@Blast545 consider using `strcpy` to copy strings, here and elsewhere

Reply:
Actually replaced this with strcpy and all the linters complain about it (windows even complain when using NOLINT). I'll set it back to memcpy again

Reply:
How did it complain? Maybe using the safer `strncpy` remains an option?